validator/src/state/merkle.rs

```diff
@@ -140,7 +140,7 @@ impl MerkleDatabase {
 
         let mut batch = Vec::with_capacity(path_addresses.len());
         // initializing this to empty, to make the compiler happy
-        let mut key_hash = String::new();
+        let mut key_hash = Vec::with_capacity(0);
```

Comment:
Why not `Vec::new()`?

Reply:
To me, Vec::new() implies that values will be pushed on to this instance.  In this case, it's a placeholder to make the compiler happy.  What is being mutated about that variable is what it points to, not the contents of the vector.  This makes that explicit.